validator/client/beacon-api/beacon_api_validator_client.go

```diff
@@ -14,24 +15,26 @@ import (
 )
 
 type beaconApiValidatorClient struct {
-	url            string
-	httpClient     http.Client
-	fallbackClient iface.ValidatorClient
+	url                     string
+	httpClient              http.Client
+	fallbackClient          iface.ValidatorClient
+	attestationDataProvider attestationDataProvider
```

Comment:
I doubt it makes sense to create providers for everything by default. We'll just have a lot of additional properties and boilerplate code which doesn't add any value. I think they are useful in cases like `genesisProvider` in your domain data PR, where one function uses another, but not as a general rule.